packages/i18n/src/create-i18n.js

```diff
@@ -74,29 +74,29 @@ const I18N_HOOK_REGEXP = /^i18n\.(n?gettext|has_translation)(_|$)/;
  * Retrieve the domain to use when calling domain-specific filters.
  */
 /**
- * @typedef {(text: string, domain?: string) => string} __
+ * @typedef {<T extends string>(text: T, domain?: string) => import('./types').TranslatableText< T >} __
```

Comment:
[There's JSdoc `@import` support](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#import), that could avoid repeating `import(…)…<T>` everywhere.